torchtext/experimental/models/xlmr_model.py

```diff
@@ -0,0 +1,126 @@
+MD5 = {'xlmr.regular': 'adf75f3d20c8a876533b206ccb3a7cb6',
+       'xlmr.base': 'abc0a28171f883c6e1b2e8cf184c1ce8'}
+
+##################################################################################
+# This part will be moved to stl-text/models folder
+
+
+class SentenceClassificationHead(nn.Module):
+    """Head for sentence-level classification."""
+
+    def __init__(self, num_labels, embed_dim=768, dropout=0.2):
+        super(SentenceClassificationHead, self).__init__()
+        self.dense = nn.Linear(embed_dim, embed_dim)
+        self.dropout = nn.Dropout(dropout)
+        self.out_proj = nn.Linear(embed_dim, num_labels)
+        self.activation = nn.Tanh()
+
+    def forward(self, input_features):
+        x = input_features[:, 0, :]  # The first token is reserved for [CLS]
+        x = self.dense(x)
+        x = self.activation(x)
+        x = self.dropout(x)
+        x = self.out_proj(x)
+        return x
+
+
+def _load_sentence_classifier(checkpoint_file='model.pt', args_file='args.json'):
+    args = load_args_from_json(args_file)
+    classifier = SentenceClassificationHead(args.num_labels, args.embed_dim, args.dropout)
+    classifier.load_state_dict(torch.load(checkpoint_file))
+    return classifier
+
+
+class TransformerEncoderSentenceClassification(nn.Module):
+    def __init__(self, transformer_encoder, classifier_head):
+        super(TransformerEncoderSentenceClassification, self).__init__()
+        self.transformer_encoder = transformer_encoder
+        self.classifier_head = classifier_head
+
+    def forward(src):
+        raise NotImplementedError("forward func has not been implemented yet.")
+
+
+def xlmr_base_sentence_classifier(root='./.model'):
+    '''
+    Examples:
+        >>> from torchtext.experimental.models import xlmr_base_sentence_classifier
+        >>> xlmr_sentence_classifier_model, xlmr_base_transform = xlmr_base_sentence_classifier()
+        >>> xlmr_base_transform('this is an example')
+        >>> tensor([  903,    83,   142, 27781])
+    '''
+    # Load pretrained XLM-R
+    tar_file = download_from_url(PRETRAINED['xlmr.base'], root=root,
+                                 hash_value=MD5['xlmr.base'], hash_type='md5')
+    checkpoint_file, tokenizer_file, vocab_file, args_file = extract_archive(tar_file, overwrite=True)
+    xlmr_model = _load_xlmr_model(checkpoint_file=checkpoint_file, args_file=args_file)
+    xlmr_transform = _load_xlmr_transform(tokenizer_file=tokenizer_file, vocab_file=vocab_file)
+
+    # Load classifier head
+    tar_file = download_from_url(TASK_PRETRAINED['xlmr.base.sentence.classifier'], root=root,
+                                 hash_value=TASK_MD5['xlmr.base.sentence.classifier'], hash_type='md5')
+    checkpoint_file, args_file = extract_archive(tar_file, overwrite=True)
+    sentence_classifier = _load_sentence_classifier(checkpoint_file=checkpoint_file, args_file=args_file)
```

Comment:
FYI: a head is tightly coupled with the "base" that comes before it. Mixing a head with the wrong base will lead to completely invalid results.

I suppose the use-case here is for finetuning on different tasks, while keeping the "base" frozen, is that the case?

Reply:
Yes. This is the use case. You have a pretrained model as the base and fine-tune the head for different task.